ui/components/app/snaps/snap-ui-renderer/snap-ui-renderer.js

```diff
@@ -72,44 +196,47 @@ const SnapUIRendererComponent = ({
 
   const backgroundColor =
     contentBackgroundColor ??
-    mapToExtensionCompatibleColor(content?.props?.backgroundColor) ??
+    mapToExtensionCompatibleColor(processedContent?.props?.backgroundColor) ??
     BackgroundColor.backgroundAlternative;
 
+  // The renderer should only have a footer if there is a default cancel action
+  // or if the footer component has been used.
+  const hasFooter =
+    onCancel || processedContent?.props?.children?.[1] !== undefined;
+
   const sections = useMemo(
     () =>
-      content &&
+      processedContent &&
       mapToTemplate({
         map: {},
-        element: content,
+        element: processedContent,
         onCancel,
         useFooter,
         promptLegacyProps,
         t,
         contentBackgroundColor: backgroundColor,
+        // We have to pass the props at the top level to ensure the children components get the correct props.
+        requireScroll,
+        isScrolledToBottom: scrollState.buttonsEnabled,
```

Comment:
We may wanna consider enabling the buttons if the user scrolls down once, e.g. not disabling if the user scrolls up again.

cc @eriknson

Reply:
Yes, leave buttons enabled if the user _has seen the content once_ regardless if they scroll up (and down, and up, etc.) again

Reply:
Done.